PCSX2-GUI: Fix GUI inconsistencies

RedDevilus opened the pull request:
### Description of Changes
<!-- Brief description or overview on what was changed in the PR -->
PCSX2-GUI: Fix Groupboxes to align on Sound and Gamepad and fix the look of DEV9
### Rationale behind Changes
<!-- Why were these changes made?  What problem does it solve / area does it improve? -->
It looks ugly
### Suggested Testing Steps
<!-- If applicable, including examples you've already tested with / recommendations for how to test further is very helpful! -->
See if nothing breaks and looks nice.

xTVaser commented on pcsx2/DEV9/ConfigUI.cpp:
```diff
@@ -184,16 +184,16 @@ class DEV9Dialog : public wxDialog
 		m_hdd_size_slider->SetPageSize(10);
-		for (int i = 15; i < HDD_MAX_GB; i += 5)
-			m_hdd_size_slider->SetTick(i);
+		for (int i = HDD_MIN_GB; i < HDD_MAX_GB; i += 5)
+			m_hdd_size_slider->SetTickFreq(10);
```
you shouldn't have to do this in a for-loop.  It looks like the original method was manually setting the ticks -- but the whole point of TickFrequency is specifying "hey i want you to put a tick every nth increment"

RedDevilus commented:
GamePad Settings (PAD):

Old:
![image](https://user-images.githubusercontent.com/24227051/140592333-d7b4837b-acbc-4564-9280-801320031762.png)

New:
![image](https://user-images.githubusercontent.com/24227051/140592047-0c145915-4b4f-44d4-bb48-60fbe0ce07a8.png)

Audio Settings (SPU2):

Old:
![image](https://user-images.githubusercontent.com/24227051/140592756-dbfc2729-f4f8-4a7c-bf42-44faae92cb00.png)

New:
![image](https://user-images.githubusercontent.com/24227051/140592360-e626e741-3f1c-4529-ac49-28a07c2e8b28.png)

Network and HDD Settings (DEV9):
Old:
![image](https://user-images.githubusercontent.com/24227051/140592368-ff873ff7-60fb-4624-8030-cf65dc5463b8.png)

New:
![image](https://user-images.githubusercontent.com/24227051/140593148-2d4c6e9d-3436-46f2-a0dd-273366e07cec.png)

RedDevilus commented:
I removed the min max labels, they only obscure the sight and could confuse people.
![image](https://user-images.githubusercontent.com/24227051/140621100-16de81aa-6f18-499b-8a28-0e6447b4f7b7.png)

Mrlinkwii commented:
![image](https://user-images.githubusercontent.com/17685135/140622449-9c576058-9430-4755-a72e-dc3cc7beff33.png)
how it looks  on linux

tellowkrinkle commented on pcsx2/DEV9/ConfigUI.cpp:
```diff
@@ -184,16 +184,15 @@ class DEV9Dialog : public wxDialog
 		auto* hdd_size_label = new wxStaticText(this, wxID_ANY, _("HDD Size (GiB):"));
 		auto* hdd_size_box = new wxBoxSizer(wxHORIZONTAL);
-		m_hdd_size_slider = new wxSlider(this, wxID_ANY, HDD_MIN_GB, HDD_MIN_GB, HDD_MAX_GB, wxDefaultPosition, wxDefaultSize, wxSL_HORIZONTAL | wxSL_MIN_MAX_LABELS | wxSL_AUTOTICKS);
+		m_hdd_size_slider = new wxSlider(this, wxID_ANY, 10, HDD_MIN_GB, HDD_MAX_GB, wxDefaultPosition, wxDefaultSize, wxSL_HORIZONTAL | wxSL_AUTOTICKS);
```
The value you replaced with `10` is the initial value of the slider.  Putting a value less than the minimum just ends up with the minimum in a more confusing way.

RedDevilus replied:
Forgot I changed the value. Thanks.

tellowkrinkle commented on pcsx2/DEV9/ConfigUI.cpp:
```diff
@@ -184,16 +184,15 @@ class DEV9Dialog : public wxDialog
-		hdd_size_box->Add(m_hdd_size_spin,   wxSizerFlags(0).Centre());
-		hdd_grid->Add(hdd_size_label, wxSizerFlags().Centre().Right());
-		hdd_grid->Add(hdd_size_box,   wxSizerFlags().Expand());
+		hdd_size_box->Add(m_hdd_size_slider, wxSizerFlags(4).Centre().Border(wxLEFT,4)); 
+		hdd_size_box->Add(m_hdd_size_spin,   wxSizerFlags(1).Centre().Border(wxLEFT,4)); 
+		hdd_grid->Add(hdd_size_label, wxSizerFlags().Centre().Right().Border(wxLEFT,4)); 
+		hdd_grid->Add(hdd_size_box,   wxSizerFlags().Expand().Border(wxLEFT,4)); 
```
Please don't apply left borders to everything.  If you wanted to add space between the slider and spin control, you can `hdd_size_box->AddSpacer(5);` between the two controls.

I also don't see much point in messing with the proportions like that, there's no reason to grow the size box because it never needs to fit more digits than `HDD_MAX_GB`.  If you're finding that it doesn't fit that, adjust the size of the box by adjusting the number in the `GetSizeFromTextSize` call.

RedDevilus replied:
Probably why Linux/Mac looks so big as I am screwing with the window properties in janky method.

Mrlinkwii commented:
![image](https://user-images.githubusercontent.com/17685135/140624889-6f4cf4b3-b7d7-4c27-9367-c9e26542b6cc.png)
looks much better :+1:

RedDevilus commented:
Hopefully, latest revision as my VS didn't seem to like this file:
![image](https://user-images.githubusercontent.com/24227051/140631063-e2bc42ad-333c-45eb-9f8d-cc1f7b023ed9.png)

RedDevilus commented:
A bit more white space above `Hard Disk Drive`. Should be fine as the final version.
![image](https://user-images.githubusercontent.com/24227051/140662626-f51c6d25-3cfc-4c50-b6c2-1cbbca36878f.png)

IlDucci commented:
Can I dare to add some two cents?

> New:
> ![image](https://user-images.githubusercontent.com/24227051/140593182-a6d8fd49-b992-4368-b304-56281711046d.png)

You might want to add a bit of space between the right side of the Interpolation dropdown box, so it matches the left side. You could also make all the three dropdown boxes of the Mixing Settings section equal in width.

RedDevilus commented:
@IlDucci 
![image](https://user-images.githubusercontent.com/24227051/140667879-e5458655-f04f-4a71-9954-af083ad8baac.png)

Looks better? (Ignore the position of OK and Cancel, haven't changed that yet)

IlDucci commented:
Yes, that's much more uniform. Thank you!

I'd think that the only thing that doesn't look uniform now would be the right side of the Volume slider (compared with the Latency one).

RedDevilus commented:
I've already fixed it:

![image](https://user-images.githubusercontent.com/24227051/140668598-12b7558c-6bc9-48df-b3b0-1c2c2bfb6ddc.png)

RedDevilus commented:
Old is left                               
New is right


Network and HDD:
![image](https://user-images.githubusercontent.com/24227051/140669371-b7e673c3-5426-4b0a-8b9c-d8891b64143e.png)

Audio Settings:
![image](https://user-images.githubusercontent.com/24227051/140669355-947fd4f3-d014-4f75-93fd-52e1eaa8db55.png)

Gamepad Settings:
![image](https://user-images.githubusercontent.com/24227051/140669435-4c895af7-06ff-4ccb-99eb-3df2c98d0fbf.png)

RedDevilus commented:
Changed the max audio latency slider from 750 to 200. Also fixed the steps (with page down/page up/arrow keys when you select the slider instead of dragging it) when you go from TimeStretch to another mode , something I forgot due to how the code is duplicated on Windows. The Linux side is much saner in SPU2 so Windows will need a rewrite in the future. + 0.5f clamps for volume aren't probably needed either.

Default on TimeStretch:
![image](https://user-images.githubusercontent.com/24227051/140680711-8b577a80-1cf3-4c1e-8891-a7d81cbfb2bd.png)

Default on Async Mix:
![image](https://user-images.githubusercontent.com/24227051/140680753-7e5cc0d7-a576-4521-8f1d-e454816f903e.png)

Mrlinkwii commented:
differences in latency  MAX value linux 


master :
![image](https://user-images.githubusercontent.com/17685135/140784751-427a8b6c-69f9-4c12-817c-743faff22a83.png)
PR :
![image](https://user-images.githubusercontent.com/17685135/140784817-502ef513-0f92-44d3-bc7f-c60ccf06fe4a.png)
